### PR TITLE
Skeleton of `archive` action

### PR DESCRIPTION
**Pull request by anaulin**
This is a for-discussion rough draft of how we could structure the `archive` implementation for https://github.com/starlinglab/starling-integrity-api/issues/67

There is a number of details to figure out, and we can use this diff to divide up tasks between ourselves.

### CODE REVIEW

**Comment by anaulin** [comment]
I started implementing some of the archival steps directly in this same PR, which helped me figure out what other questions to ask (see my questions in Slack and also TODOs in this diff).

We probably don't want to implement ALL the things in this one PR and then merge it, though. We could either:
(a) merge this PR as it stands right now (it doesn't really "work", but it has working tests and doesn't break anything outside of archival)
(b) extract parts of this code into reviewable + mergeable chunks (e.g. the FileUtil changes could be its own PR)
(c) something else?

@makeworld-the-better-one @benhylau do you have a preference between the choices above?

File: starlingcaptureapi/file_util.py [hunk 1]
```diff
@@ -56,3 +60,53 @@ def digest_sha256(self, file_path):
...
+
+    @staticmethod
+    def make_zip(filenames: list[str], out_file: str):
+        """Makes a zip file containing the given list of files.
+
+        Args:
+            filenames: list of full filenames to include in zip
+            out_file: full path to output zip file
+        """
+        with zipfile.ZipFile(out_file, "w") as zipf:
```
Comment (benhylau) [inline starlingcaptureapi/file_util.py:108]: This could raise right?

File: starlingcaptureapi/file_util.py [hunk 2]
```diff
@@ -56,3 +60,53 @@ def digest_sha256(self, file_path):
...
+            for filename in filenames:
+                # This defaults to having an archive name that matches the given filename,
```
Comment (benhylau) [inline starlingcaptureapi/file_util.py:110]: By `given filename` do you mean `out_file`?

File: tests/context.py [hunk 3]
```diff
@@ -8,5 +8,6 @@
 from starlingcaptureapi import claim
 from starlingcaptureapi import config
 from starlingcaptureapi import exif
+from starlingcaptureapi import file_util
```
Comment (benhylau) [inline tests/context.py:11]: This is growing~ 😅

`s/starlingcaptureapi/starlingintegrityeapi`

**Comment by benhylau** [comment]
Let's walk thru the PR at our call and then do (a).

**Pull request merged**